tests/components/samsungtv/test_media_player.py

```diff
@@ -170,61 +104,20 @@ def mock_now():
 
 async def setup_samsungtv(hass, config):
     """Set up mock Samsung TV."""
-    await async_setup_component(hass, "media_player", config)
+    await async_setup_component(hass, SAMSUNGTV_DOMAIN, config)
     await hass.async_block_till_done()
 
 
-async def test_setup_with_mac(hass, remote):
+async def test_setup_with_turnon(hass, remote):
     """Test setup of platform."""
     await setup_samsungtv(hass, MOCK_CONFIG)
     assert hass.states.get(ENTITY_ID)
 
 
-async def test_setup_duplicate(hass, remote, caplog):
-    """Test duplicate setup of platform."""
-    DUPLICATE = {DOMAIN: [MOCK_CONFIG[DOMAIN], MOCK_CONFIG[DOMAIN]]}
-    await setup_samsungtv(hass, DUPLICATE)
-    assert "Ignoring duplicate Samsung TV fake" in caplog.text
-
-
-async def test_setup_without_mac(hass, remote):
+async def test_setup_without_turnon(hass, remote):
     """Test setup of platform."""
-    await setup_samsungtv(hass, MOCK_CONFIG_NOMAC)
-    assert hass.states.get(ENTITY_ID_NOMAC)
-
-
-async def test_setup_discovery(hass, remote):
-    """Test setup of platform with discovery."""
-    hass.async_create_task(
-        async_load_platform(
-            hass, DOMAIN, SAMSUNGTV_DOMAIN, MOCK_CONFIG_DISCOVERY, {DOMAIN: {}}
-        )
-    )
-    await hass.async_block_till_done()
-    state = hass.states.get(ENTITY_ID_DISCOVERY)
-    assert state
-    assert state.name == "fake_discovery (fake_model)"
-    entity_registry = await hass.helpers.entity_registry.async_get_registry()
-    entry = entity_registry.async_get(ENTITY_ID_DISCOVERY)
-    assert entry
-    assert entry.unique_id == "fake_uuid"
-
-
-async def test_setup_discovery_prefix(hass, remote):
-    """Test setup of platform with discovery."""
-    hass.async_create_task(
-        async_load_platform(
-            hass, DOMAIN, SAMSUNGTV_DOMAIN, MOCK_CONFIG_DISCOVERY_PREFIX, {DOMAIN: {}}
-        )
-    )
-    await hass.async_block_till_done()
-    state = hass.states.get(ENTITY_ID_DISCOVERY_PREFIX)
-    assert state
-    assert state.name == "fake_discovery_prefix (fake_model_prefix)"
-    entity_registry = await hass.helpers.entity_registry.async_get_registry()
-    entry = entity_registry.async_get(ENTITY_ID_DISCOVERY_PREFIX)
-    assert entry
-    assert entry.unique_id == "fake_uuid_prefix"
+    await setup_samsungtv(hass, MOCK_CONFIG_NOTURNON)
+    assert hass.states.get(ENTITY_ID_NOTURNON)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4795320)